### PR TITLE
feat: custom directory and selecting of an existing directory

### PR DESCRIPTION
**Pull request by 0tuedon**

### CODE REVIEW

**Comment by 0tuedon** [comment]
This PR references #131  and commit history issues have been fixed

**Comment by 0tuedon** [comment]
@Extheoisah  @Emmanuel-Develops , An update has been made to fit PR #168  for the directories

**Comment by 0tuedon** [comment]
https://github.com/bitcointranscripts/transcription-review-front-end/assets/90271995/3703ab7d-a4db-4e5f-a0b6-68110819c60d

**Review by Emmanuel-Develops — CHANGES_REQUESTED** [review]
A couple of logic that needs to be implemented
- [ ]  selecting a nested directory should update the path, e.g advancing-bitcoin/2021 and not only 2021
- [ ] users can save their selected directories on the save button
- [ ] if user has selected an arbitrary/custom directory, append that directory to the list
- [ ] when navigating with the back button, go back one level up the directory

**Comment by 0tuedon** [comment]
> A couple of logic that needs to be implemented
> 
> * [ ]   selecting a nested directory should update the path, e.g advancing-bitcoin/2021 and not only 2021
> * [ ]  users can save their selected directories on the save button
> * [ ]  if user has selected an arbitrary/custom directory, append that directory to the list
> * [ ]  when navigating with the back button, go back one level up the directory

Okay, on this

**Comment by 0tuedon** [comment]
Hey @Emmanuel-Develops, i have made some updates to this PR, you could go through it and LMKWYT.

**Comment by Emmanuel-Develops** [comment]
> Hey @Emmanuel-Develops, i have made some updates to this PR, you could go through it and LMKWYT.

Looks Good, Nice!

**Review by Emmanuel-Develops — COMMENTED** [review]
One last thing, can you add filtering to the list

This is good so that when user want to create an arbitrary directory they can see a list of similar ones, incase it already exists. 
Or just generally quickly find a specific directory.

**Comment by Emmanuel-Develops** [comment]
> One last thing, can you add filtering to the list. We discussed it in the last meeting (I forgot to include it in the checklist above)
> 
> This is good so that when user want to create an arbitrary directory they can see a list of similar ones, incase it already exists. Or just generally quickly find a specific directory.

Let's make this a separate PR, as an additional feature on the nested directories

File: src/types/api/dir.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  size: number;
+  url: string;
+  html_url: string;
+  git_url: string;
+  download_url: null;
+  type: string;
+  _links: Links;
+};
+
+export type Links = {
```
Comment (Emmanuel-Develops) [inline src/types/api/dir.ts:14]: and this?

File: src/types/api/dir.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+export type DirectoryRes = {
```
[the remark this answers is not in the document]
Comment (0tuedon) [reply src/types/api/dir.ts:1]: nope

File: src/types/api/dir.ts (hunk 1 above)
Comment (0tuedon) [reply src/types/api/dir.ts:14]: not in use

**Comment by 0tuedon** [comment]
@Emmanuel-Develops  could you check this out again?

**Comment by Emmanuel-Develops** [comment]
> @Emmanuel-Develops could you check this out again?

alright, checking it out!

**Review by Emmanuel-Develops — APPROVED** [review]
Looks good!
Other changes will be in a separate pr

**Pull request merged**